driver-core/src/main/com/mongodb/connection/ClusterSettings.java

```diff
@@ -254,12 +278,30 @@ public Builder applyConnectionString(final ConnectionString connectionString) {
             maxWaitQueueSize(waitQueueMultiple * maxSize);
 
             if (connectionString.getServerSelectionTimeout() != null) {
-                serverSelectionTimeout(connectionString.getServerSelectionTimeout(), TimeUnit.MILLISECONDS);
+                serverSelectionTimeout(connectionString.getServerSelectionTimeout(), MILLISECONDS);
+            }
+            if (connectionString.getLocalThreshold() != null) {
+                localThreshold(connectionString.getLocalThreshold(), MILLISECONDS);
             }
-
             return this;
         }
 
+        private ServerSelector unpackServerSelector(final ServerSelector serverSelector) {
+            if (serverSelector instanceof CompositeServerSelector) {
+                List<ServerSelector> serverSelectors = ((CompositeServerSelector) serverSelector).getServerSelectors();
+                return new CompositeServerSelector(serverSelectors.subList(1, serverSelectors.size()));
```

Comment:
We know based on how packServerSelector is written that if it's a Composite it will have exactly two items: the latency minimizing selector and the application-provided selector.  So just grab the first one, which should be the application-provided selector (see comment below)

Reply:
Done